### DISCUSSION TITLE
Confusing panic

### DISCUSSION THREAD

**Discussion by clcc2019** [grafana/mimir#8645] [Help and support]
I updated from 2.10.5 to 2.11 and above and experienced intermittent distributor panics, a sudden increase in distributor memory causing pod ooms, and write latency of 1s-2s, which was <100ms before the upgrade

Can't understand the improved performance and stability optimizations in the new version

**Comment by dimitarvdimitrov** [comment]
whelp that doesn't sound great :( can you share the actual panic? A profile of a distributor with higher memory would also help (`curl distributor-address:8080/debug/pprof/heap > profile.pprof` and then upload profile.pprof)
